src/plugins/github/graphql.js

```diff
@@ -340,6 +340,108 @@ function* continuationsFromReview(
   }
 }
 
+/**
+ * Merge structured data from the given `source` into a given subpath of
+ * the `destination`. The original inputs are not modified.
+ *
+ * Arrays are merged by concatenation. Objects are merged by recursively
+ * merging each key. Primitives are merged by replacement (the
+ * destination is simply overwritten with the source).
+ *
+ * See test cases for examples.
+ *
+ * NOTE: The type of `source` should be the same as the type of
+ *
+ *     destination[path[0]][path[1]]...[path[path.length - 1]],
+ *
+ * but this constraint cannot be expressed in Flow so we just use `any`.
+ */
+export function merge<T>(
+  destination: T,
+  source: any,
+  path: $ReadOnlyArray<string | number>
+): T {
+  function isObject(x) {
+    return !Array.isArray(x) && typeof x === "object" && x != null;
+  }
+  if (path.length > 0) {
+    const key = path[0];
+    function checkKey(key: string | number, destination: Object | Array<any>) {
+      if (!(key in destination)) {
+        const keyText = JSON.stringify(key);
+        const destinationText = JSON.stringify(destination);
+        throw new Error(
+          `Key ${keyText} not found in destination: ${destinationText}`
+        );
+      }
+    }
+    if (typeof key === "number") {
```

Comment:
shouldn't this block be a part of checkKey?

Reply:
Ah, I see, that's because this is actually opening the "handle number key block" - I misread it as just being associated with another check. Another point for my "can this be separated a bit more clearly for readability" request, though.

Reply:
Noted. I'll split them out locally and see what I think.